### ISSUE TITLE
Clean Template parser to try to consume Quarkus Qute

### ISSUE THREAD

**Issue by angelozerr** [redhat-developer/quarkus-ls#473]
Clean Template parser to try to consume Quarkus Qute

**Comment by angelozerr** [comment]
@mkouba I know it's a big work, but do you think you could be interested with my Qute parser?

Here the source code of Qute parser https://github.com/redhat-developer/quarkus-ls/tree/master/qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/parser

and the tests https://github.com/redhat-developer/quarkus-ls/tree/master/qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/parser

I know that it requires a lot ofrefactoring onmy side but I would like to know if you think you could be interested?

**Comment by mkouba** [comment]
@angelozerr I totally agree that it would make sense to have a shared parser implementation based on AST. However, I will not be able to push this forward in the near future. That said, I'd be happy to help you make your parser _usable_ by the Qute core ;-)

**Comment by angelozerr** [comment]
> @angelozerr I totally agree that it would make sense to have a shared parser implementation based on AST. 

Glad you like this idea:)

> However, I will not be able to push this forward in the near future. That said, I'd be happy to help you make your parser usable by the Qute core ;-)

No problem I understand that it's not your priority but I create this issue to do that in the future. The main improvement that I see in Qute core is that you could manage robust offsets for error validation.
